tests/transforms/test_convert_pdl_to_pdl_interp.py

```diff
@@ -3747,3 +3747,329 @@ def map_rewrite_value(val: SSAValue) -> SSAValue:
 
     # Verify no mapping was added
     assert attr_op.output not in rewrite_values
+
+
+def test_generate_operation_result_type_rewriter_strategy1_all_resolvable():
+    """Test _generate_operation_result_type_rewriter Strategy 1: all types resolvable"""
+    from xdsl.dialects.builtin import ModuleOp
+    from xdsl.ir import Block, Region
+    from xdsl.transforms.convert_pdl_to_pdl_interp.conversion import MatcherGenerator
+
+    # Setup
+    matcher_body = Region([Block(arg_types=(pdl.OperationType(),))])
+    matcher_func = pdl_interp.FuncOp(
+        "matcher", ((pdl.OperationType(),), ()), region=matcher_body
+    )
+    rewriter_module = ModuleOp([])
+    generator = MatcherGenerator(matcher_func, rewriter_module)
+
+    # Create rewriter region with an operation that has resolvable types
+    rewriter_region = Region([Block()])
+    rewriter_block = rewriter_region.first_block
+    with ImplicitBuilder(rewriter_block):
+        type1 = pdl.TypeOp(i32).result
+        type2 = pdl.TypeOp(f32).result
+        op_to_create = pdl.OperationOp("test.op", type_values=(type1, type2))
+
+    rewrite_values: dict[SSAValue, SSAValue] = {}
+
+    # Map the type values
+    type_block = Block()
+    type1_arg = type_block.insert_arg(pdl.TypeType(), 0)
+    type2_arg = type_block.insert_arg(pdl.TypeType(), 1)
+    rewrite_values[type1] = type1_arg
+    rewrite_values[type2] = type2_arg
+
+    def map_rewrite_value(val: SSAValue) -> SSAValue:
+        return rewrite_values.get(val, val)
+
+    types_list: list[SSAValue] = []
+
+    # Call method
+    has_inferred = generator._generate_operation_result_type_rewriter(  # pyright: ignore[reportPrivateUsage]
+        op_to_create, map_rewrite_value, types_list, rewrite_values
+    )
+
+    # Verify Strategy 1 was used: all types resolved
+    assert has_inferred is False
+    assert len(types_list) == 2
+    assert types_list[0] == type1_arg
+    assert types_list[1] == type2_arg
```

Comment:
pytest gives nicer error messages when comparing lists
```suggestion
    assert types_list == [type1_arg, type2_arg]
```